packages/qwik-city/middleware/node/http.ts

```diff
@@ -7,6 +8,9 @@ import type {
 const { ORIGIN, PROTOCOL_HEADER, HOST_HEADER = 'host' } = process.env;
 
 function getOrigin(req: IncomingMessage) {
+  if (req instanceof Http2ServerRequest) {
```

Comment:
Not sure it's ok to reoverride completely the other logic, since it's useful for proxy support

Reply:
I think the responsibility of `getOrigin()` is limited to obtain the origin string from the request object with no side effect.
I guess the proxy support can be provided in `getUrl()`, because the proxy needs other parts of URL not only the origin.

Reply:
The origin might not be the one from the request, in some cases the backend is behind a reverse proxy and the origin of the request is not relevant at all.

Reply:
Imo we should respect the HOST_HEADER when it's not Host, and the PROTOCOL_HEADER

Reply:
I see, you meant not about the forward proxy but the reverse proxy.
Yes, I will fix it :)

Reply:
@manucorporat Updated PR.
Changed the default value of the HOST_HEADER to be conditional to http version, if HTTP/2 then `:authority` else `host`.